### DISCUSSION TITLE
Relationship of two parts of model output

### DISCUSSION THREAD

**Discussion by Kieran31** [ultralytics/yolov5#7774] [Q&A] [answered]
https://github.com/ultralytics/yolov5/blob/4870064629e76d9d387578c562a292cc680fa05f/val.py#L206
There are two parts in the model ouput. If my training size is 512x512 and val batch size is 1, `out` has shape `[1,16128,6]`, `train_out[0]` has shape `[1,3,64,64,6]`, `train_out[1]` has shape `[1,3,32,32,6]`, `train_out[2]` has shape `[1,3,16,16,6]`. Clearly, the first dimension is the batch size and the last dimension 6 means `[xywh, cls, conf]`. 
I'm curious about the other dimensions. Since `16128 == 3*64*64 + 3*32*32 + 3*16*16`, is there any relationship between `out` and `train_out`? Can we convert `train_out` to `out`?

**Comment by glenn-jocher** [comment, marked as answer]
@Kieran31 as the variable names state the first output is used for training. dimensions are batch anchor x y feature

**Reply by Kieran31** [reply]
Thank you. Do you mean `train_out`?
Are `train_out` and `out` literally independent?

**Reply by glenn-jocher** [reply]
@Kieran31 they are variations of the same features. See Detect() module for details of model outputs:
https://github.com/ultralytics/yolov5/blob/4870064629e76d9d387578c562a292cc680fa05f/models/yolo.py#L37-L76
